flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV310.java

```diff
@@ -111,4 +112,28 @@ public void makeSpecFromName(Map<String, String> partSpec, Path currPath) {
 		}
 	}
 
+	@Override
+	public Set<String> getNotNullColumns(IMetaStoreClient client, Configuration conf, String dbName, String tableName) {
+		try {
+			Method method = getMetaStoreUtilsClass().getDeclaredMethod("getDefaultCatalog", Configuration.class);
+			String hiveDefaultCatalog = (String) method.invoke(null, conf);
+			Class requestClz = Class.forName("org.apache.hadoop.hive.metastore.api.NotNullConstraintsRequest");
+			Object request = requestClz.getDeclaredConstructor(String.class, String.class, String.class)
+					.newInstance(hiveDefaultCatalog, dbName, tableName);
+			method = client.getClass().getDeclaredMethod("getNotNullConstraints", requestClz);
```

Comment:
Can we have some reflection utils like: `Object invoke(Object obj, String method, Object... args)`.
In this method, we can get base `class` from `obj`, and get parameter classes from `args`.

Reply:
Added this util method but we actually cannot get base class from obj, and parameter classes from args. Because the target method can be a static method in which case obj is null, and args can be sub-classes of parameters declared in method signature.